dialer_test.go

```diff
@@ -286,25 +309,31 @@ func TestConnectState(main *testing.T) {
 
 		ctrl := gomock.NewController(t)
 		defer ctrl.Finish()
-
+		// TODO
 		l := logger.NewTest()
+		unreadyCh := make(chan error, 10)
 
 		ctx, cancelFunc := context.WithCancel(context.Background())
-		defer cancelFunc()
 
 		dialer, err := amqpextra.NewDialer(
 			amqpextra.WithURL("amqp://rabbitmq.host"),
 			amqpextra.WithAMQPDial(amqpDialStub(fmt.Errorf("the error"))),
 			amqpextra.WithRetryPeriod(time.Millisecond*150),
+			amqpextra.WithUnreadyCh(unreadyCh),
 			amqpextra.WithLogger(l),
 			amqpextra.WithContext(ctx),
 		)
 		require.NoError(t, err)
 
 		time.Sleep(time.Millisecond * 100)
-		assertUnready(t, dialer, "the error")
+		assertUnready(t, unreadyCh, amqp.ErrClosed.Error())
 
 		cancelFunc()
+
+		time.Sleep(time.Millisecond * 100)
+
+		assertUnready(t, unreadyCh, "the error")
```

Comment:
do lines 333-335 after assertUnready(amqp.ErrClose) and before cancelFunc()

Reply:
+